include/zephyr/bluetooth/buf.h

```diff
@@ -48,15 +50,60 @@ enum bt_buf_type {
 	BT_BUF_ISO_OUT = BIT(4),
 	/** Incoming ISO data */
 	BT_BUF_ISO_IN = BIT(5),
-	/** H:4 data */
-	BT_BUF_H4 = BIT(6),
 };
 
-/** @brief This is a base type for bt_buf user data. */
-struct bt_buf_data {
-	uint8_t type;
+/** Direction of HCI packets */
+enum bt_buf_dir {
+	BT_BUF_IN,
+	BT_BUF_OUT,
```

Comment:
From the point of view of the host? 

We could consider making the directions absolute rather than relative like we have for ISO data paths: 

```c
#define BT_HCI_DATAPATH_DIR_HOST_TO_CTLR 0x00
#define BT_HCI_DATAPATH_DIR_CTLR_TO_HOST 0x01
```

The same argument can be applied for the ACL and ISO IN/OUT, but if we modify one, we should probably modify all, which seems outside the scope of this PR

Reply:
> From the point of view of the host?

From the point of view of `BT_BUF_*_IN/OUT values when converting from a H:4 type (which don't convey any direction info by themselves).

Reply:
I added a clarifying sentence to the docstring above the enum definition.

Reply:
Also worth noting that this entire public header will hopefully disappear soon, so I wouldn't put too much effort into trying to micro-optimize it :)